src/providers/WorkflowCore.QueueProviders.RabbitMQ/Services/RabbitMQProvider.cs

```diff
@@ -7,23 +7,30 @@
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
 using WorkflowCore.Interface;
 using WorkflowCore.Models;
+using WorkflowCore.QueueProviders.RabbitMQ.Interfaces;
 
 namespace WorkflowCore.QueueProviders.RabbitMQ.Services
 {
 #pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
     public class RabbitMQProvider : IQueueProvider
     {
-        private readonly IConnectionFactory _connectionFactory;
+        private readonly IRabbitMqQueueNameProvider _queueNameProvider;
+        private readonly RabbitMqConnectionFactory _rabbitMqConnectionFactory;
+        private readonly IServiceProvider _serviceProvider;
+        
         private IConnection _connection = null;
         private static JsonSerializerSettings SerializerSettings = new JsonSerializerSettings() { TypeNameHandling = TypeNameHandling.All };
 
         public bool IsDequeueBlocking => false;
 
-        public RabbitMQProvider(IConnectionFactory connectionFactory)
+        public RabbitMQProvider(IServiceProvider serviceProvider)
```

Comment:
I think we should inject the actual dependencies rather than the IoC container?

Reply:
Explicit dependencies added